src/main/java/ir/sahab/cleanup/Cleanups.java

```diff
@@ -63,27 +69,31 @@ public void doAllQuietly() {
         }
     }
 
+    /**
+     * Does all cleanup operations and if there is an exception on any operation, throws the first
+     * exception after trying for all operations. Probably the first error was the cause of the
+     * problem for the rest of the operations and therefore the most important error to investigate
+     * the problem.
+     * @throws IOException if there is an exception on any operation, wraps it in {@link IOException}
+     *         and returns it. Most of the time, the {@code close()} function in which {@link IOException}
+     *         occurs is used as clean-up operation. For this reason, we wrap the catched error with it.
+     */
```

Comment:
Replace:

any operaion, ... > any operation. It contains the original exception as its cause. We choose to throw an exception of type IOException, because on of the main use cases of this class is in implementation of close() methods of {@link Closeable} objects where you want to delegate the close operations to the objects downstream of the chain.